client/scripts/models/SearchQuery.ts

```diff
@@ -0,0 +1,109 @@
+/* eslint-disable prefer-template */
+export enum SearchScope {
+    "Members" = "Members",
+    "Communities" = "Communities",
+    "Proposals" = "Proposals",
+    "Threads" = "Threads",
+    "Replies" = "Replies",
+    "All" = "All"
+}
+
+export enum SearchSort {
+    "Best" = "Best",
+    "Newest" = "Newest",
+    "Oldest" = "Oldest"
+}
+export interface SearchParams {
+    communityScope?: string;
+    chainScope?: string;
+    isSearchPreview?: boolean;
+    searchScope?: Array<SearchScope>;
+    sort?: SearchSort;
+    resultSize?: number;
+}
+export default class SearchQuery implements SearchParams {
+    public searchTerm: Lowercase<string>;
+    public communityScope?: string;
+    public chainScope?: string;
+    public isSearchPreview?: boolean;
+    public searchScope: Array<SearchScope>;
+    public sort: SearchSort;
+
+    constructor(searchTerm: string, params?: SearchParams){
```

Comment:
Love that you set up a SearchQuery class, really cleans things up.

For these declarations, you might try using TS `?.` syntax, e.g. `params?.sort || SearchSort.Best`, `= params?.communityScope`, `!!params.isSearchPreview`. But this is fine too, pretty legible.

Reply:
Changed, thanks for the TS tip!